src/core/services/theming/theming.js

```diff
@@ -528,7 +528,7 @@ function generateAllThemes($injector) {
   if (generateOnDemand) return;
 
   angular.forEach(THEMES, function(theme) {
-    if (!GENERATED[theme.name]) {
+    if (!GENERATED[theme.name] && !(theme.name !== $mdTheming.defaultTheme() && theme.name === 'default')) {
```

Comment:
it's just that 
```js
!(theme.name !== $mdTheming.defaultTheme() && theme.name === 'default')
```

doesn't makes sense, you are checking if `theme.name` isn't equals to something AND equals to something else, if you know that you expect it to be `default` than why do you even bother to check if it's not equals to something else